docs/IE_DG/Model_caching_overview.md

```diff
@@ -1,65 +1,136 @@
+
+### Set the “CACHE_DIR” config option to enable model caching
+
+To enable model caching, the application must specify the folder where to store cached blobs. It can be done using [IECore.set_config](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.set_config).
+
+``` python
+from openvino.inference_engine import IECore
+
+ie = IECore()
+ie.set_config(config={"CACHE_DIR": path_to_cache}, device_name=device)
+net = ie.read_network(model=path_to_xml_file)
+exec_net = ie.load_network(network=net, device_name=device)
+```
+
+With this code, if a device supports the Import/Export network capability, a cached blob is automatically created inside the path_to_cache directory `CACHE_DIR` config is set to the Core object. If device does not support Import/Export capability, cache is just not created and no error is thrown
+
+Depending on your device, total time for loading network on application startup can be significantly reduced. Please also note that very first [IECore.load_network](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.load_network) (when the cache is not yet created) takes slightly longer time to ‘export’ the compiled blob into a cache file.
+
+![caching_enabled]
+
+
+### Even Faster: Use IECore.load_network(path_to_xml_file)
+
+In some cases, applications do not need to customize inputs and outputs every time. These applications always call [IECore.read_network](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.read_network), then `IECore.load_network(model=path_to_xml_file)` and may be further optimized. For such cases, it's more convenient to load the network in a single call to `ie.load_network()`
+A model can be loaded directly to the device, with model caching enabled:
+
+``` python
+from openvino.inference_engine import IECore
+
+ie = IECore()
+ie.set_config(config={"CACHE_DIR" : path_to_cache}, device_name=device)
+ie.load_network(network=path_to_xml_file, device_name=device)
+```
+
+![caching_times]
+
+### Advanced Examples
+
+Not every device supports network import/export capability, enabling of caching for such devices does not have any effect. To check in advance if a particular device supports model caching, your application can use the following code:
+
+```python
+all_metrics = ie.get_metric(device_name=device, metric_name="SUPPORTED_METRICS")
+# Find the 'IMPORT_EXPORT_SUPPORT' metric in supported metrics
+allows_caching = "IMPORT_EXPORT_SUPPORT" in all_metrics
```

Comment:
```suggestion
allows_caching = "IMPORT_EXPORT_SUPPORT" in all_metrics and ie.get_metric(device_name=device, metric_name="IMPORT_EXPORT_SUPPORT")
```

Reply:
Existence of this metric in 'Supported metrics' list doesn't necessarily guarantee that device supports caching. Plugin also needs to return 'True' value for this metric.
Even though existing plugins which declare 'IMPORT_EXPORT_SUPPORT' in supported metrics, always return 'true', it is allowed to return 'False' if plugin decides so.
Also if you check 'C++' snippet - there are 2 checks for this:
```
// Find 'IMPORT_EXPORT_SUPPORT' metric in supported metrics
auto it = std::find(keys.begin(), keys.end(), METRIC_KEY(IMPORT_EXPORT_SUPPORT));

// If metric 'IMPORT_EXPORT_SUPPORT' exists, check it's value
auto cachingSupported = (it != keys.end()) && ie.GetMetric(deviceName, METRIC_KEY(IMPORT_EXPORT_SUPPORT)).as<bool>();
```

It may be confusing for user that C++ needs value check, but Python code does not

Reply:
Thanks @nosovmik . I agree that similarity between C++ and Python is good, but to me it is now confusing that "SUPPORTED_METRICS" does not necessarily mean that that metric is supported. What would it mean if a plugin declares IMPORT_EXPORT_SUPPORT but still returns False with get_metric? Why would it then declare IMPORT_EXPORT_SUPPORT?

Reply:
@nosovmik could you please add your comments?

Reply:
Hi @helena-intel 
Sorry for delay with reply. I agree that extra check looks redundant, however there is a kind of API limitation - if plugin declares some metric - this metric shall return some value. So we agreed to have 'boolean' metric which returns 'true', and in this case now plugin has a possibility to return 'false' and it shall be treated correctly.
My suggestion is that you can create an issue to simplify the API so that we can re-visit current implementation and find best solution. From one hand, extra check looks redundant for end user. From another hand, it is 'advanced api', so it shall not be a problem for advanced users to add extra check if their application needs this functionality